packages/sso-express/src/controllers.ts

```diff
@@ -121,7 +126,8 @@ export const authCallbackController =
 
     try {
       const tokenSet = await client.callback(
-        client.metadata.redirect_uris[0],
+        options.getRedirectUri(new URL(client.metadata.redirect_uris[0]), req)
+          .href,
```

Comment:
Will `req` have the expected query parameters here, e.g. `redirectTo` ? Should we save the redirect uri in the session instead of calling `getRedirectUri` twice?

Reply:
Good idea!

Reply:
Neither option here would prevent an attacker who intercepted the authorization request from forging a token request with the correct `redirect_uri`.
That is the role of [PKCE](https://datatracker.ietf.org/doc/html/rfc7636) and the `code_verifier` parameter below